### ISSUE TITLE
United States 2022 - New Years Day is Incorrect

### ISSUE THREAD

**Issue by mv5903** [nager/Nager.Date#419]
### Affected country

United States

### Affected public holiday

New Year's Day

### Source of the information

Just wrong day, 12-31-2021. https://date.nager.at/api/v3/PublicHolidays/2022/us

**Comment by tinohager** [comment]
What is the correct rule?

https://github.com/nager/Nager.Date/blob/master/src/Nager.Date/PublicHolidays/UnitedStatesProvider.cs


> New Year's Day, Juneteenth, Independence Day, Veterans Day, and Christmas Day are observed on the same calendar date each year, irrespective of the day of the week. When a holiday falls on a **Saturday**, federal employees who work Monday to Friday **observe the holiday on the previous Friday**. Federal employees who work on Saturday observe the holiday on Saturday and, for them, Friday is a regular work day. Holidays that fall on a Sunday are observed by federal workers the following Monday. The other holidays always fall on a specific day of the week.[14]
https://en.wikipedia.org/wiki/Federal_holidays_in_the_United_States

**Comment by Codain** [comment]
As a non-american I understand the Wikipedia as the following:

1. The above holidays are on fixed calendar dates.
2. If it falls on a Saturday and if you are working in a Federal agency/administration then you will have an additional holiday on Friday if you were not supposed to work on this day (but New Year's Eve is still on Saturday).
3. If it falls on a Sunday and if you are working in a Federal agency/administration then you will have an additional holiday on Monday (but New Year's Eve is still on Sunday).

Consequently I would write the following rules for New Year's Day (can push a Pull Request if needed):

```
var newYearsDay = new DateTime(year, 1, 1);
items.Add(new PublicHoliday(newYearsDay, "New Year's Day", "New Year's Day", countryCode));
items.Add(new PublicHoliday(new DateTime(year, 1, 1).Shift(saturday => saturday.AddDays(-1), sunday => sunday.AddDays(+1)), "New Year's Day compensation (Federal)", "New Year's Day compensation (Federal)", countryCode, null, null, PublicHolidayType.Authorities));
```

The type of the holiday is not ideal but I picked the closest one I found. The name of the holiday can be changed (but idea is to reflect the applicability to certain type of administrations). This way you have the real Holiday and the possible additional day off.
